Build failure: python3Packages.pythonocc-core

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

    nix-build '<nixpkgs>' -A python3Packages.pythonocc-core

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/285748882

### Relevant log output

```console
/build/source/build/Unix/x86_64-Release-/Graphic3dPYTHON_wrap.cxx:3859:38: error: cannot bind non-const lvalue reference of type 'Graphic3d_SequenceOfHClipPlane::Iterator&' to an rvalue of type 'std::remove_reference<Graphic3d_SequenceOfHClipPlane::Iterator&>::type' {aka 'Graphic3d_SequenceOfHClipPlane::Iterator'}
 3859 | # define SWIG_STD_MOVE(OBJ) std::move(OBJ)
      |                             ~~~~~~~~~^~~~~
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.13.0-rc6, NixOS, 25.05 (Warbler), 25.05.20250116.5df4362`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"nixos-20.03pre194293.2436c27541b"`
 - nixpkgs: `/home/aengelen/nixpkgs`


### Notify maintainers

@gebner 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/pythonocc-core/default.nix
+++ b/pkgs/development/python-modules/pythonocc-core/default.nix
@@ -14,19 +14,20 @@
   libXi,
   libXmu,
   opencascade-occt,
+  numpy,
   rapidjson,
   swig,
 }:
 
 stdenv.mkDerivation rec {
   pname = "pythonocc-core";
-  version = "7.6.2";
+  version = "7.8.1.1";
 
   src = fetchFromGitHub {
     owner = "tpaviot";
     repo = "pythonocc-core";
     tag = version;
-    hash = "sha256-45pqPQ07KYlpFwJSAYVHbzuqDQTbAvPpxReal52DCzU=";
+    hash = "sha256-0o2PQEN0/Z7FUPZEo2HxFFa+mN2bZnYI++HVu4ONpNA=";
   };
 
   postPatch = ''
@@ -50,12 +51,14 @@ stdenv.mkDerivation rec {
     libXmu
     libXi
     fontconfig
+    numpy
     rapidjson
   ] ++ lib.optionals stdenv.hostPlatform.isDarwin [ Cocoa ];
 
   cmakeFlags = [
     "-Wno-dev"
     "-DPYTHONOCC_INSTALL_DIRECTORY=${placeholder "out"}/${python.sitePackages}/OCC"
+    "-DPYTHONOCC_MESHDS_NUMPY=on"
   ];
 
   passthru = {

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -13617,7 +13617,6 @@ self: super: with self; {
     inherit (pkgs) fontconfig rapidjson;
     inherit (pkgs.xorg) libX11 libXi libXmu libXext;
     inherit (pkgs.darwin.apple_sdk.frameworks) Cocoa;
-    opencascade-occt = pkgs.opencascade-occt_7_6;
   });
 
   python-olm = callPackage ../development/python-modules/python-olm { };

